programs/protocol_event/src/instructions/create_event.rs

```diff
@@ -0,0 +1,164 @@
+use crate::error::EventError;
+use crate::state::event::Event;
+use anchor_lang::prelude::*;
+
+#[derive(AnchorSerialize, AnchorDeserialize, Debug, Clone, PartialEq)]
+pub struct CreateEventInfo {
+    pub slug: String,
+    pub name: String,
+    pub participants: Vec<u16>,
+    pub expected_start_timestamp: i64,
+    pub actual_start_timestamp: Option<i64>,
+    pub actual_end_timestamp: Option<i64>,
+}
+
+pub fn create(
+    event: &mut Event,
+    event_info: CreateEventInfo,
+    authority: Pubkey,
+    payer: Pubkey,
+    category: Pubkey,
+    event_group: Pubkey,
+) -> Result<()> {
+    require!(
+        event_info.name.len() <= Event::MAX_NAME_LENGTH,
+        EventError::MaxStringLengthExceeded,
+    );
+    require!(
+        event_info.slug.len() <= Event::MAX_SLUG_LENGTH,
+        EventError::MaxStringLengthExceeded,
+    );
+
+    event.authority = authority;
+    event.payer = payer;
+
+    event.category = category;
+    event.event_group = event_group;
+
+    event.active = false;
+
+    event.slug = event_info.slug;
+    event.name = event_info.name;
+
+    event.participants = event_info.participants;
```

Comment:
I don't think we'd be able to breach it within a single transaction's size limit, but perhaps we should be checking participants length also.

Reply:
I also wonder whether we should verify that the participant indices in the vec are at least within the bounds of `[0, category.participant_count)`?

Reply:
I've created a pattern of seperate `validate_x` methods for these, makes unit tests easier too